testing/endtoend/evaluators/fee_recipient.go

```diff
@@ -0,0 +1,140 @@
+package evaluators
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/config/params"
+	validator_service_config "github.com/prysmaticlabs/prysm/config/validator/service"
+	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/testing/endtoend/helpers"
+	e2e "github.com/prysmaticlabs/prysm/testing/endtoend/params"
+	"github.com/prysmaticlabs/prysm/testing/endtoend/policies"
+	"github.com/prysmaticlabs/prysm/testing/endtoend/types"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+var FeeRecipientIsPresent = types.Evaluator{
+	Name:       "Fee_Recipient_Is_Present_%d",
+	Policy:     policies.AfterNthEpoch(helpers.BellatrixE2EForkEpoch),
+	Evaluation: feeRecipientIsPresent,
+}
+
+func feeRecipientIsPresent(conns ...*grpc.ClientConn) error {
+	conn := conns[0]
+	client := ethpb.NewBeaconChainClient(conn)
+	chainHead, err := client.GetChainHead(context.Background(), &emptypb.Empty{})
+	if err != nil {
+		return errors.Wrap(err, "failed to get chain head")
+	}
+	req := &ethpb.ListBlocksRequest{QueryFilter: &ethpb.ListBlocksRequest_Epoch{Epoch: chainHead.HeadEpoch.Sub(1)}}
+	blks, err := client.ListBeaconBlocks(context.Background(), req)
+	if err != nil {
+		return errors.Wrap(err, "failed to list blocks")
+	}
+
+	rpcclient, err := rpc.DialHTTP(fmt.Sprintf("http://127.0.0.1:%d", e2e.TestParams.Ports.Eth1RPCPort))
+	if err != nil {
+		return err
+	}
+	defer rpcclient.Close()
+	web3 := ethclient.NewClient(rpcclient)
+	ctx := context.Background()
+
+	var configFiles []*validator_service_config.ProposerSettingsPayload
+	testNetDir := e2e.TestParams.TestPath + "/proposer-settings"
```

Comment:
reading directories via an evaluator seems ugly. The main purpose should be to just fetch api data via the grpc endpoints and use that to make a decision.

Reply:
The prepare proposer api is only one directional so I can check the original file configs that are passed in through it. There is no get setting.

Reply:
If you make the mapping of (pubkey => fee-recipient) stateless, you can avoid checking the original file config. 
Ex: If we use the bls key to determine the fee recipient , all we need to do is find the bls public key for a block proposer
and use that to derive the fee recipient. You can fetch this all via the api ( block's have the proposer index in them)

Reply:
ah I see what you're saying , thanks for the clarification I'll do this then.